ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,94 @@
+import axios from 'axios'
+import FormData from 'form-data'
+import { randomUUID } from 'crypto'
+import { HttpService } from '@nestjs/axios'
+import { ConfigService } from '@nestjs/config'
+import { Injectable } from '@nestjs/common'
+import baseLogger from '../logger'
+import { CanvasUserLoginEmail } from '../canvas/canvas.interfaces'
+import { Config } from '../config'
+import { HttpMethod } from '../api/api.utils'
+import { InvitationAPIError } from './invitation.errors'
+import { lastValueFrom } from 'rxjs'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (
+    private readonly configService: ConfigService<Config, true>,
+    private readonly httpService: HttpService
+  ) {
+    const invitationConfig = configService.get('invitation', { infer: true })
+    this.url = invitationConfig.apiURL
+    this.entityID = invitationConfig.apiEntityID
+    this.sponsorName = invitationConfig.apiSponsorName
+    this.key = invitationConfig.apiKey
+    this.secret = invitationConfig.apiSecret
+  }
+
+  async sendInvitations (users: CanvasUserLoginEmail[]): Promise<string> {
+    if (users.length == 0) {
+      throw new InvitationAPIError('Argument "users" array is empty.')
```

Comment:
@lsloan Currently when I am trying to run this locally using `api/sections/570673/enrollExternal` with input/Post body like this 
```
{
  "users": [
    {
      "email": "simple@gmail.com",
      "givenName": "simple",
      "surname": "argument",
      "type": "DesignerEnrollment"
    },
{
      "email": "simpledouble@gmail.com",
      "givenName": "double",
      "surname": "argument",
      "type": "TeacherEnrollment"
    }
  ]
}
```

I get a response as 

```{
  "statusCode": 500,
  "message": "Internal server error"
}
```

I see the server logs as below

```
[Nest] 100  - 02/07/2022, 8:33:52 PM   ERROR [ExceptionsHandler] An error occurred while sending invitations: Argument "users" array is empty.
ccm_web     | InvitationAPIError: An error occurred while sending invitations: Argument "users" array is empty.
ccm_web     |     at CirrusInvitationService.<anonymous> (/base/server/src/invitation/cirrus-invitation.service.ts:38:13)
ccm_web     |     at Generator.next (<anonymous>)
ccm_web     |     at /base/server/src/invitation/cirrus-invitation.service.ts:17:71
ccm_web     |     at new Promise (<anonymous>)
ccm_web     |     at __awaiter (/base/server/src/invitation/cirrus-invitation.service.ts:13:12)
ccm_web     |     at CirrusInvitationService.sendInvitations (/base/server/src/invitation/cirrus-invitation.service.ts:45:16)
ccm_web     |     at APIService.<anonymous> (/base/server/src/api/api.service.ts:178:71)
ccm_web     |     at Generator.next (<anonymous>)
ccm_web     |     at fulfilled (/base/server/src/api/api.service.ts:14:58)
ccm_web     |     at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Am I testing this correctly?**

Reply:
Probably.

However, since @ssciolla's review comments, I've made changes to fix some things, but some other things are now not working.  It's not working for me, either.

If I can, I'll try to clarify the current state of the PR by removing your name from the list of reviewers. When the code is once again in a reviewable state, I'll indicate to you that it is again ready to review.